src/ResourceManager/Dns/Commands.Dns/Zones/SetAzureDnsZone.cs

```diff
@@ -53,17 +80,46 @@ public override void ExecuteCmdlet()
             DnsZone result = null;
             DnsZone zoneToUpdate = null;
 
-            if (this.ParameterSetName == "Fields")
+            if (this.ParameterSetName == FieldsIdsParameterSetName || this.ParameterSetName == FieldsObjectsParameterSetName)
             {
-                zoneToUpdate = new DnsZone
+                if (this.Name.EndsWith("."))
                 {
-                    Name = this.Name,
-                    ResourceGroupName = this.ResourceGroupName,
-                    Etag = "*",
-                    Tags = this.Tag,
-                };
+                    this.Name = this.Name.TrimEnd('.');
```

Comment:
Would this have previously caused the cmdlet to fail?

Reply:
This check was already present in the file, on line 76.